src/vis/vis.js

```diff
@@ -360,6 +383,17 @@ var VisModel = Backbone.Model.extend({
       })
       .flatten()
       .value();
+  },
+
+  _getFilters: function () {
+    return this._dataviewsCollection.reduce(function (filters, dataview) {
```

Comment:
All we care about here is if any of the dataviews have a filter, so I'd try to do something simpler like:

```
_isAnyDataviewFiltered: function () {
  return this._dataviewsCollection.any(function (dataview) {
    return dataview.isActive(); // !dataview.isEmpty()
  });
}
```

If you feel 💪 , perhaps we could create a `DataviewsCollection` so we can put this logic there...